### ISSUE TITLE
Providing methods to simplify behaviour after delegating state

### ISSUE THREAD

**Issue by paulszefer** [paulszefer/Ecosystem-Simulation#4]
Hi Chris,
I have implemented a Health class to delegate health-related state and behaviour from Guppy. One such delegated method was incrementAge. Should I still provide an incrementAge method in Guppy that simply passes on the request to its health profile?

For example, previous requests to increment the age of a Guppy looked like:
`guppy.incrementAge()`
Now, the request looks as follows:
`guppy.getHealth().incrementAge()`

Is the reduction in chained method calls preferred over saving a few lines of code in Guppy that would have simply passed on the incrementAge request?

**Comment by chris-thompson** [comment]
Yes, this is legit.  Take a peek at Demeter's rule.  It may help explain why we do this.

**Comment by paulszefer** [comment]
From [Java Examples on the Law of Demeter](https://alvinalexander.com/java/java-law-of-demeter-java-examples)

> In short, the Law of Demeter aims to keep you from doing things like this:
> 
> `objectA.getObjectB().doSomething();`
> 
> or even worse, this:
> 
> `objectA.getObjectB().getObjectC().doSomething();`

Whoops.
I'll have to do another pass to fix these.

**Comment by chris-thompson** [comment]
It's pretty neat.  It decreases how tightly coupled our classes are.  Something you may wish to try: there are lots of plugins for Eclipse that will convert your code to a UML diagram.  Do find one and try it out.  You will want to reduce the connections between classes.  We can definitely look at this together when we meet!
